server/src/test/java/org/opensearch/common/blobstore/stream/read/listener/FilePartWriterTests.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.blobstore.stream.read.listener;
+
+import org.opensearch.common.io.InputStreamContainer;
+import org.opensearch.test.OpenSearchTestCase;
+import org.opensearch.threadpool.TestThreadPool;
+import org.opensearch.threadpool.ThreadPool;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.opensearch.common.blobstore.stream.read.listener.ListenerTestUtils.TestCompletionListener;
+
+public class FilePartWriterTests extends OpenSearchTestCase {
+
+    private Path path;
+    private static ThreadPool threadPool;
+
+    @BeforeClass
+    public static void setup() {
+        threadPool = new TestThreadPool(FilePartWriterTests.class.getName());
+    }
+
+    @AfterClass
+    public static void cleanup() {
+        threadPool.shutdown();
+    }
+
+    @Before
+    public void init() throws Exception {
+        path = createTempDir("FilePartWriterTests");
+    }
+
+    public void testFilePartWriter() throws Exception {
+        Path segmentFilePath = path.resolve(UUID.randomUUID().toString());
+        int contentLength = 100;
+        int partNumber = 1;
+        InputStream inputStream = new ListenerTestUtils.TestInputStream(contentLength);
+        InputStreamContainer inputStreamContainer = new InputStreamContainer(inputStream, inputStream.available(), 0);
+        AtomicBoolean anyStreamFailed = new AtomicBoolean();
+        TestCompletionListener<Integer> fileCompletionListener = new TestCompletionListener<>();
+
+        FilePartWriter filePartWriter = new FilePartWriter(
+            partNumber,
+            inputStreamContainer,
+            segmentFilePath,
+            anyStreamFailed,
+            fileCompletionListener
+        );
+        threadPool.executor(ThreadPool.Names.GENERIC).submit(filePartWriter).get();
```

Comment:
I don't think you need a thread pool here and can replace these lines with `filePartWriter.run()`